src/arr/compiler/cli-repl.js

```diff
@@ -28,6 +28,9 @@
             console.log('Received "exit" event from repl!');
             restarter.resume("repl done");
           });
+          replServer.on('reset', () => {
+            restartInteractions.app("");
```

Comment:
This should need to be `runtime.runThunk(function() { return restartInteractions.app(""); }, function(result) { /* probably just ignore the result here */ });`

This is because `restartInteractions`, in principle, could capture the stack and need to be restarted in the middle. In this case, I can't seem to make this break, but I think it's a useful comment regardless.

Reply:
Done! Thanks for the tip. I'm still trying to wrap my head around all of the Pyret internals.